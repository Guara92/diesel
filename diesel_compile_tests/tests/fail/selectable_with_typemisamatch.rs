extern crate diesel;

use diesel::prelude::*;

table! {
    users {
        id -> Integer,
        name -> Text,
    }
}

#[derive(Selectable, Queryable)]
#[diesel(table_name = users)]
struct User {
    id: String,
    name: i32,
}


fn main() {
    let mut conn = PgConnection::establish("...").unwrap();

    users::table.select(User::as_select()).load(&mut conn).unwrap();


}
