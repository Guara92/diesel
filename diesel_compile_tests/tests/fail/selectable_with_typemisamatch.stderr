error[E0277]: the trait bound `i32: FromSql<diesel::sql_types::Text, Mysql>` is not satisfied
  --> tests/fail/selectable_with_typemisamatch.rs:16:11
   |
16 |     name: i32,
   |           ^^^ the trait `FromSql<diesel::sql_types::Text, Mysql>` is not implemented for `i32`
   |
   = help: the following other types implement trait `FromSql<A, DB>`:
             <f32 as FromSql<diesel::sql_types::Float, Mysql>>
             <f32 as FromSql<diesel::sql_types::Float, Pg>>
             <f32 as FromSql<diesel::sql_types::Float, Sqlite>>
             <f64 as FromSql<diesel::sql_types::Double, Mysql>>
             <f64 as FromSql<diesel::sql_types::Double, Pg>>
             <f64 as FromSql<diesel::sql_types::Double, Sqlite>>
             <i16 as FromSql<diesel::sql_types::SmallInt, Mysql>>
             <i16 as FromSql<diesel::sql_types::SmallInt, Pg>>
           and 13 others
   = note: required because of the requirements on the impl of `diesel::Queryable<diesel::sql_types::Text, Mysql>` for `i32`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: the trait bound `i32: FromSql<diesel::sql_types::Text, Sqlite>` is not satisfied
  --> tests/fail/selectable_with_typemisamatch.rs:16:11
   |
16 |     name: i32,
   |           ^^^ the trait `FromSql<diesel::sql_types::Text, Sqlite>` is not implemented for `i32`
   |
   = help: the following other types implement trait `FromSql<A, DB>`:
             <f32 as FromSql<diesel::sql_types::Float, Mysql>>
             <f32 as FromSql<diesel::sql_types::Float, Pg>>
             <f32 as FromSql<diesel::sql_types::Float, Sqlite>>
             <f64 as FromSql<diesel::sql_types::Double, Mysql>>
             <f64 as FromSql<diesel::sql_types::Double, Pg>>
             <f64 as FromSql<diesel::sql_types::Double, Sqlite>>
             <i16 as FromSql<diesel::sql_types::SmallInt, Mysql>>
             <i16 as FromSql<diesel::sql_types::SmallInt, Pg>>
           and 13 others
   = note: required because of the requirements on the impl of `diesel::Queryable<diesel::sql_types::Text, Sqlite>` for `i32`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: the trait bound `i32: FromSql<diesel::sql_types::Text, Pg>` is not satisfied
  --> tests/fail/selectable_with_typemisamatch.rs:16:11
   |
16 |     name: i32,
   |           ^^^ the trait `FromSql<diesel::sql_types::Text, Pg>` is not implemented for `i32`
   |
   = help: the following other types implement trait `FromSql<A, DB>`:
             <f32 as FromSql<diesel::sql_types::Float, Mysql>>
             <f32 as FromSql<diesel::sql_types::Float, Pg>>
             <f32 as FromSql<diesel::sql_types::Float, Sqlite>>
             <f64 as FromSql<diesel::sql_types::Double, Mysql>>
             <f64 as FromSql<diesel::sql_types::Double, Pg>>
             <f64 as FromSql<diesel::sql_types::Double, Sqlite>>
             <i16 as FromSql<diesel::sql_types::SmallInt, Mysql>>
             <i16 as FromSql<diesel::sql_types::SmallInt, Pg>>
           and 13 others
   = note: required because of the requirements on the impl of `diesel::Queryable<diesel::sql_types::Text, Pg>` for `i32`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: the trait bound `*const str: FromSql<diesel::sql_types::Integer, Mysql>` is not satisfied
  --> tests/fail/selectable_with_typemisamatch.rs:15:9
   |
15 |     id: String,
   |         ^^^^^^ the trait `FromSql<diesel::sql_types::Integer, Mysql>` is not implemented for `*const str`
   |
   = help: the following other types implement trait `FromSql<A, DB>`:
             <*const str as FromSql<diesel::sql_types::Text, Mysql>>
             <*const str as FromSql<diesel::sql_types::Text, Pg>>
             <*const str as FromSql<diesel::sql_types::Text, Sqlite>>
             <std::string::String as FromSql<ST, DB>>
             <std::string::String as FromSql<TimestamptzSqlite, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Date, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Time, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Timestamp, Sqlite>>
   = note: required because of the requirements on the impl of `FromSql<diesel::sql_types::Integer, Mysql>` for `std::string::String`
   = note: required because of the requirements on the impl of `diesel::Queryable<diesel::sql_types::Integer, Mysql>` for `std::string::String`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: the trait bound `*const str: FromSql<diesel::sql_types::Integer, Sqlite>` is not satisfied
  --> tests/fail/selectable_with_typemisamatch.rs:15:9
   |
15 |     id: String,
   |         ^^^^^^ the trait `FromSql<diesel::sql_types::Integer, Sqlite>` is not implemented for `*const str`
   |
   = help: the following other types implement trait `FromSql<A, DB>`:
             <*const str as FromSql<diesel::sql_types::Text, Mysql>>
             <*const str as FromSql<diesel::sql_types::Text, Pg>>
             <*const str as FromSql<diesel::sql_types::Text, Sqlite>>
             <std::string::String as FromSql<ST, DB>>
             <std::string::String as FromSql<TimestamptzSqlite, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Date, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Time, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Timestamp, Sqlite>>
   = note: required because of the requirements on the impl of `FromSql<diesel::sql_types::Integer, Sqlite>` for `std::string::String`
   = note: required because of the requirements on the impl of `diesel::Queryable<diesel::sql_types::Integer, Sqlite>` for `std::string::String`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable

error[E0277]: the trait bound `*const str: FromSql<diesel::sql_types::Integer, Pg>` is not satisfied
  --> tests/fail/selectable_with_typemisamatch.rs:15:9
   |
15 |     id: String,
   |         ^^^^^^ the trait `FromSql<diesel::sql_types::Integer, Pg>` is not implemented for `*const str`
   |
   = help: the following other types implement trait `FromSql<A, DB>`:
             <*const str as FromSql<diesel::sql_types::Text, Mysql>>
             <*const str as FromSql<diesel::sql_types::Text, Pg>>
             <*const str as FromSql<diesel::sql_types::Text, Sqlite>>
             <std::string::String as FromSql<ST, DB>>
             <std::string::String as FromSql<TimestamptzSqlite, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Date, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Time, Sqlite>>
             <std::string::String as FromSql<diesel::sql_types::Timestamp, Sqlite>>
   = note: required because of the requirements on the impl of `FromSql<diesel::sql_types::Integer, Pg>` for `std::string::String`
   = note: required because of the requirements on the impl of `diesel::Queryable<diesel::sql_types::Integer, Pg>` for `std::string::String`
   = help: see issue #48214
   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
